Currently, we are having an "importing module" problem. Such as 

```
 Cannot find module 'components/common/ui/SelectMenu' from 'LoginForm.jsx'
```

Rob has a way to fix this. I am currently investigating a more generic way of fixing this issue. Maybe having a `JSON` file and let the `jest in `package.json` call the new JSON file.

--- a/jest-config.json
+++ b/jest-config.json
@@ -1,0 +1,15 @@
+{
+  "verbose": true,
+  "unmockedModulePathPatterns": [
+    "<rootDir>/node_modules/react"
+  ],
+  "transform": {".*": "<rootDir>/node_modules/babel-jest"},
+  "moduleFileExtensions": [
+    "js",
+    "jsx"
+  ],
+  "moduleNameMapper": {
+    "^modals$": "<rootDir>/troposphere/static/js/modals.js",
+    "^components/common/ui/SelectMenu$": "<rootDir>/troposphere/static/js/components/common/ui/SelectMenu.jsx"
+  }
+}

--- a/jest-config.json
+++ b/jest-config.json
@@ -10,6 +10,7 @@
   ],
   "moduleNameMapper": {
     "^modals$": "<rootDir>/troposphere/static/js/modals.js",
-    "^components/common/ui/SelectMenu$": "<rootDir>/troposphere/static/js/components/common/ui/SelectMenu.jsx"
+    "^components/common/ui/SelectMenu$": "<rootDir>/troposphere/static/js/components/common/ui/SelectMenu.jsx",
+    "^context":  "<rootDir>/troposphere/static/js/context.js"
   }
 }

--- a/package.json
+++ b/package.json
@@ -21,7 +21,7 @@
     "build": "webpack --progress --profile --colors --bail --",
     "watch": "webpack --progress --profile --colors --bail --watch -d --",
     "jenkins": "webpack --profile --colors --",
-    "test": "jest",
+    "test": "jest --config jest-config.json",
     "stats": "webpack --profile --json > stats.json",
     "lint": "eslint troposphere/static/js",
     "travis-lint": "eslint -c .eslintrc.travis --no-eslintrc troposphere/static/js",

--- a/troposphere/static/js/components/login/LoginHeader.spec.jsx
+++ b/troposphere/static/js/components/login/LoginHeader.spec.jsx
@@ -1,0 +1,21 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import renderer from 'react-test-renderer';
+
+describe("Login Header", () => {
+  var ReactTestUtils = require('react-addons-test-utils'),
+      LoginHeader = React.createFactory(require('./LoginHeader.jsx'));
+
+  it("should render without any error", () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<LoginHeader />, div);
+  });
+  it("snapshot test", () => {
+    const tree = renderer.create(
+      <LoginHeader />
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+});

